core/services/keystore/workflow_encryption.go

```diff
@@ -0,0 +1,162 @@
+}
+
+func (ks *workflowEncryption) GetAll() (keys []workflowencryptionkey.Key, _ error) {
+	ks.lock.RLock()
+	defer ks.lock.RUnlock()
+	if ks.isLocked() {
+		return nil, ErrLocked
+	}
+	for _, key := range ks.keyRing.WorkflowEncryption {
+		keys = append(keys, key)
+	}
+	return keys, nil
+}
+
+func (ks *workflowEncryption) Create(ctx context.Context) (workflowencryptionkey.Key, error) {
+	ks.lock.Lock()
+	defer ks.lock.Unlock()
+	if ks.isLocked() {
+		return workflowencryptionkey.Key{}, ErrLocked
+	}
+	// Ensure you can only have one WorkflowEncryption at a time.
+	if len(ks.keyRing.WorkflowEncryption) > 0 {
+		return workflowencryptionkey.Key{}, ErrWorkflowEncryptionKeyExists
+	}
+
+	key, err := workflowencryptionkey.New()
+	if err != nil {
+		return workflowencryptionkey.Key{}, err
+	}
+	return key, ks.safeAddKey(ctx, key)
+}
+
+func (ks *workflowEncryption) Add(ctx context.Context, key workflowencryptionkey.Key) error {
+	ks.lock.Lock()
+	defer ks.lock.Unlock()
+	if ks.isLocked() {
+		return ErrLocked
+	}
+	if len(ks.keyRing.WorkflowEncryption) > 0 {
+		return ErrWorkflowEncryptionKeyExists
+	}
+	return ks.safeAddKey(ctx, key)
+}
+
+func (ks *workflowEncryption) Delete(ctx context.Context, id string) (workflowencryptionkey.Key, error) {
+	ks.lock.Lock()
+	defer ks.lock.Unlock()
+	if ks.isLocked() {
+		return workflowencryptionkey.Key{}, ErrLocked
+	}
+	key, err := ks.getByID(id)
+	if err != nil {
+		return workflowencryptionkey.Key{}, err
+	}
+
+	err = ks.safeRemoveKey(ctx, key)
+
+	return key, err
+}
+
+func (ks *workflowEncryption) Import(ctx context.Context, keyJSON []byte, password string) (workflowencryptionkey.Key, error) {
+	ks.lock.Lock()
+	defer ks.lock.Unlock()
+	if ks.isLocked() {
+		return workflowencryptionkey.Key{}, ErrLocked
+	}
+	key, err := workflowencryptionkey.FromEncryptedJSON(keyJSON, password)
+	if err != nil {
+		return workflowencryptionkey.Key{}, errors.Wrap(err, "WorkflowEncryptionKeyStore#ImportKey failed to decrypt key")
+	}
+	if _, found := ks.keyRing.WorkflowEncryption[key.ID()]; found {
+		return workflowencryptionkey.Key{}, fmt.Errorf("key with ID %s already exists", key.ID())
+	}
+	return key, ks.keyManager.safeAddKey(ctx, key)
+}
+
+func (ks *workflowEncryption) Export(id string, password string) ([]byte, error) {
+	ks.lock.RLock()
+	defer ks.lock.RUnlock()
+	if ks.isLocked() {
+		return nil, ErrLocked
+	}
+	key, err := ks.getByID(id)
+	if err != nil {
+		return nil, err
+	}
+	return key.ToEncryptedJSON(password, ks.scryptParams)
+}
+
+// EnsureKey verifies whether the WorkflowEncryption key has been seeded, if not, it creates it.
+func (ks *workflowEncryption) EnsureKey(ctx context.Context) error {
+	ks.lock.Lock()
+	defer ks.lock.Unlock()
+	if ks.isLocked() {
+		return ErrLocked
+	}
+
+	if len(ks.keyRing.WorkflowEncryption) > 0 {
+		return nil
+	}
+
+	key, err := workflowencryptionkey.New()
+	if err != nil {
+		return err
+	}
+
+	ks.logger.Infof("Created Encryption key with ID %s", key.ID())
```

Comment:
is there a reason why logging happens only at this line and not anywhere else?

Reply:
Im also replicating how other keystore log the creation of the key here. It makes sense to me given is done after the `.New()`,  is there another place you would log it?

Reply:
ok, let's do it in the same way other keystore do it. In my opinion, such logging does not seem consistent, but we can keep it as it is now.